ros/src/waypoint_loader/waypoint_loader.py

```diff
@@ -72,11 +72,14 @@ def decelerate(self, waypoints):
         return waypoints
 
     def publish(self, waypoints):
-        lane = Lane()
-        lane.header.frame_id = '/world'
-        lane.header.stamp = rospy.Time(0)
-        lane.waypoints = waypoints
-        self.pub.publish(lane)
+        rate = rospy.Rate(1)
+        while not rospy.is_shutdown():
```

Comment:
Why do you publish the way points in that while loop?
They are needed only once by the other modules.
The changes here make `tl_detector.py` load the way points and find the stop line endlessly.
As a result everything else is stuck, the car is not able drive and no prediction is made.
On the other hand if I revert to the previous version of that file (`waypoint_loader.py`), everything works fine (car driving and traffic light predictions).

Reply:
Oddly enough the changes in waypoint_loader.py do no cause the problem mentioned above when using the branch of PR #12.

Reply:
Good catch ! I didn't mean to do that. Correcting right now.